kibbeh/src/ui/NativeCheckbox.tsx

```diff
@@ -1,121 +1,92 @@
-import React, { Children } from "react";
+import React, { useState } from "react";
 
-export interface ToggleContext {
-  title?: string;
-  description?: string;
-  on?: boolean;
-  toggle?: (s: boolean) => any;
-  onClick?: any; // for now :)
-  className?: string;
-  children?: React.ReactNode;
+export interface SwitchProps {
+  checked: boolean;
 }
-export const MyToggleContext = React.createContext<ToggleContext>({});
 
-export const Toggle: React.FC<ToggleContext> = ({ children }) => {
-  const [on, setOn] = React.useState(false);
-  const toggle = () => setOn(!on);
+export const Switch: React.FC<SwitchProps> = ({ checked }) => {
   return (
-    <MyToggleContext.Provider value={{ on, toggle }}>
-      {children}
-    </MyToggleContext.Provider>
+    <div
+      className={`w-5.5 h-4 relative rounded-full border px-1 transition duration-400 ease-in-out-hard ${
+        checked ? "border-primary-100" : "border-primary-300"
+      }`}
+    >
+      <div
+        className={`w-2 h-2  rounded-full transition duration-400 ease-in-out-hard absolute top-2/4 left-2/4 transform -translate-y-1/2 ${
+          checked
+            ? "translate-x-0 bg-primary-100"
+            : "-translate-x-full bg-primary-300"
+        }`}
+      />
+    </div>
   );
 };
 
-const ToggleSwitch: React.FC<ToggleContext> = ({
-  on,
+export interface NativeCheckboxProps {
+  title: string;
+  subtitle: string;
+  onClick?: (num: number | undefined) => void;
+  checked?: boolean;
+  num?: number;
+}
+
+export const NativeCheckbox: React.FC<NativeCheckboxProps> = ({
+  title,
+  subtitle,
   onClick,
-  className = "",
-  ...props
+  checked = false,
+  num,
 }) => {
   return (
-    <label aria-label={"Toggle"} aria-hidden="false">
-      <input
-        // className="form-checkbox"
-        id="toggle"
-        type="checkbox"
-        checked={on}
-        onChange={() => {}}
-        onClick={onClick}
-        data-testid="toggle-input"
-      />
-      <span
-        className={on ? "text-primary-100" : "text-primary-300"}
-        {...props}
-      />
-    </label>
+    <button
+      className="w-full flex px-3 py-2 bg-primary-900 rounded-8 justify-between group"
+      onClick={onClick ? () => onClick(num) : undefined}
+    >
+      <div className="flex flex-col items-start">
+        <div className="font-bold group-hover:text-primary-100 text-primary-100 transition duration-200">
+          {title}
+        </div>
```

Comment:
currently, group-hover doesn't have any effect on the title. Also, line 46 should be
 ```className={`check ? "font-bold text-primary-100 transition duration-200" : "font-bold group-hover:text-primary-100 text-primary-300 transition duration-200">`}```

When the switch is checked it'll allow the user to visually see when the title turns white, otherwise, keep the hove effect primary-100 and the title primary-300 when the check is false.

Reply:
@ajmnz applied the fix